Source/Core/UICommon/CommandLineParse.cpp

```diff
@@ -31,21 +31,27 @@ class CommandLineConfigLayerLoader final : public Config::ConfigLayerLoader
     if (!audio_backend.empty())
       m_values.emplace_back(Config::MAIN_DSP_HLE.location, ValueToString(audio_backend == "HLE"));
 
+    if (batch)
+      m_values.emplace_back(Config::MAIN_RENDER_TO_MAIN.location, ValueToString(false));
+
     // Arguments are in the format of <System>.<Section>.<Key>=Value
-    for (const auto& arg : args)
+    if (!args.empty())
```

Comment:
This feels a bit redundant, for loop's body will just never be entered if the container is empty.

Reply:
I actually get a crash without this and I'm not sure why.

Reply:
With `args` being `std::list`!?!? Sounds like something is severely wrong and you may just be working around some major breakage. Race somewhere?

Reply:
I've made a proper fix by not calling `options.all("config")` when not allowed.